assembl/graphql/schema.py

```diff
@@ -1,81 +1,439 @@
+
+    subject = graphene.String(lang=graphene.String())
+    body = graphene.String(lang=graphene.String())
+    sentiment_counts = graphene.Field(SentimentCounts)
+
+    def resolve_subject(self, args, context, info):
+        subject = resolve_langstring(self.get_subject(), args.get('lang'))
+        return subject
+
+    def resolve_body(self, args, context, info):
+        body = resolve_langstring(self.get_body(), args.get('lang'))
+        return body
 
+    def resolve_sentiment_counts(self, args, context, info):
+        sentiment_counts = self.sentiment_counts
+        return SentimentCounts(
+            dont_understand=sentiment_counts['dont_understand'],
+            disagree=sentiment_counts['disagree'],
+            like=sentiment_counts['like'],
+            more_info=sentiment_counts['more_info'],
+        )
 
-class User(AgentProfile):
 
+class Post(SecureObjectType, SQLAlchemyObjectType):
     class Meta:
-        model = models.User
-        interfaces = (URINode, )
-        exclude_fields = ('password')
+        model = models.Post
+        interfaces = (Node, PostInterface)
+        only_fields = ('id', )  # inherits fields from Post interface only
+
+
+class PropositionPost(Post):
+    class Meta:
+        model = models.PropositionPost
+        interfaces = (Node, PostInterface)
+        only_fields = ('id', )  # inherits fields from Post interface only
+
+
+class PostUnion(SQLAlchemyUnion):
+    class Meta:
+        types = (PropositionPost, Post)
+        model = models.Post
+
+    @classmethod
+    def resolve_type(cls, instance, context, info):
+        if isinstance(instance, graphene.ObjectType):
+            return type(instance)
+        elif isinstance(instance, models.PropositionPost): # must be above Post
+            return PropositionPost
+        elif isinstance(instance, models.Post):
+            return Post
+
+
+class PostConnection(graphene.Connection):
+    class Meta:
+        node = PostUnion
+
+
+class Video(graphene.ObjectType):
+    title = graphene.String()
+    description = graphene.String()
+    html_code = graphene.String()
+
+
+class Idea(SecureObjectType, SQLAlchemyObjectType):
+    class Meta:
+        model = models.Idea
+        interfaces = (Node, )
+        only_fields = ('id', )
+
+
+class Question(SecureObjectType, SQLAlchemyObjectType):
+    class Meta:
+        model = models.Question
+        interfaces = (Node, )
+        only_fields = ('id', )
+
+    title = graphene.String(lang=graphene.String())
+    posts = graphene.List(PropositionPost)
+
+    def resolve_title(self, args, context, info):
+        title = resolve_langstring(self.title, args.get('lang'))
+        return title
+
+    def resolve_posts(self, args, context, info):
+        # TODO
+        return []
+
+
+class Thematic(SecureObjectType, SQLAlchemyObjectType):
+    class Meta:
+        model = models.Thematic
+        interfaces = (Node, )
+        only_fields = ('id', 'identifier')
```

Comment:
identifier is used as an identifier for the phase, it's the same behavior that for the timeline.